src/ESPEasyRules.ino

```diff
@@ -304,59 +304,58 @@ void replace_EventValueN_Argv(String& line, const String& argString, unsigned in
   }
 }
 
-void process_substring(String& line) {
-  // Syntax like 12345[substring:8:12:ANOTHER HELLO WORLD]67890
-  int startIndex;
-  while ((startIndex = line.indexOf(F("[substring:"))) != -1) {  
-    String substCmd = line.substring(startIndex, line.indexOf("]", startIndex) + 1);
-    int idxStart = substCmd.indexOf(":") + 1;
-    int idxEnd = substCmd.indexOf(":", idxStart);
-    String substStart = substCmd.substring(idxStart, idxEnd);
-    idxStart = idxEnd + 1;
-    idxEnd = substCmd.indexOf(":", idxStart);
-    String substEnd = substCmd.substring(idxStart , idxEnd);
-    idxStart = idxEnd + 1;
-    idxEnd = substCmd.indexOf("]", idxStart);
-    String stringValue = substCmd.substring(idxStart, idxEnd);
-  
-    line.replace(substCmd, stringValue.substring(substStart.toInt(), substEnd.toInt()));
-  }
-}
 
-void process_strtol(String& line) {
-  // Syntax like 1234[strtol:16:38]7890
+void process_internal(String &line, String cmd_s) {
   int startIndex;
-  while ((startIndex = line.indexOf(F("[strtol:"))) != -1) {    
-    String strolCmd = line.substring(startIndex, line.indexOf("]", startIndex) + 1);
-    int idxStart = strolCmd.indexOf(":") + 1;
-    int idxEnd = strolCmd.indexOf(":", idxStart);
-    String base = strolCmd.substring(idxStart, idxEnd);
-    idxStart = idxEnd + 1;
-    idxEnd = strolCmd.indexOf("]", idxStart);
-    String stringValue = strolCmd.substring(idxStart, idxEnd);
-       
-    line.replace(strolCmd, String(strtol(stringValue.c_str(), NULL, base.toInt())));   
-  }
-}
-
-void process_div100ths(String& line) {
-  // Syntax like XXX[div100ths:24:256]XXX
-  int startIndex;
-  while ((startIndex = line.indexOf(F("[div100ths:"))) != -1) {
-    String divCmd = line.substring(startIndex, line.indexOf("]", startIndex) + 1);
-    int idxStart = divCmd.indexOf(":") + 1;
-    int idxEnd = divCmd.indexOf(":", idxStart);
-    String dividend = divCmd.substring(idxStart, idxEnd);
-    idxStart = idxEnd + 1;
-    idxEnd = divCmd.indexOf("]", idxStart);
-    String divisor = divCmd.substring(idxStart, idxEnd);
-    float val = (100.0 * dividend.toInt()) / (1.0 * divisor.toInt());
+  int iarg1, iarg2;
+  while ((startIndex = line.indexOf(cmd_s)) != -1) {
+    String fullCommand = line.substring(startIndex, line.indexOf("]", startIndex) + 1);
```

Comment:
Maybe also check for the presence of `]` in the string?
For example:
```c++
  while ((startIndex = line.indexOf(cmd_s)) != -1 && line.indexOf("]", startIndex) != -1) {
```

Reply:
Added the check.